docs/tutorials/basics/70_blocks.md

```diff
@@ -0,0 +1,570 @@
+
+So first, we will declare a *block* as a method parameter: it should be placed last and the parameter's name should be prefixed by `&`. Then we can use `yield` as before.
+
+```crystal-play
+def transform_string(word, &block)
+  block_result = yield word
+  puts block_result
+end
+
+transform_string "crystal" do |word|
+  word.capitalize
+end
+```
+
+Now, we can add type restrictions to our method's parameters:
+
+```crystal-play
+def transform_string(word : String, &block : String -> String)
+  block_result = yield word
+  puts block_result
+end
+
+transform_string "crystal" do |word|
+  word.capitalize
+end
+```
+
+Let's focus on the block's type `&block : String -> String`. What we are saying is that the block will receive a `String` and return a `String`.
+
+So, if the block tries to return an `Int` the compiler will say:
+
+```crystal
+def transform_string(word : String, &block : String -> String)
+  block_result = yield word
+  puts block_result
+end
+
+transform_string "crystal" do |word|
+  42 # Error: expected block to return String, not Int32
+end
+```
+
+Or if we try to give an `Array(String)` as an input to the block, the compiler will say:
+
+```crystal
+def transform_string(word : String, &block : String -> String)
+  block_result = yield [word] # Error: argument #1 of yield expected to be String, not Array(String)
+  puts block_result
+end
+
+transform_string "crystal" do |word|
+  word.capitalize
+end
+```
+
+## Alternative syntaxes
+
+We almost at the end of this section. Let's see some other interesting ways of writing *blocks*.
+
+### Curly braces
+
+Another way of defining a *block* is using `{...}` instead of `do ... end`. Here is one of the examples we have already seen but written with *curly braces syntax*:
+
+```crystal-play
+def other_method
+  yield 42, "Hello", [1, "Crystal", 3]
+end
+
+other_method { |n, s, arr|
+  puts "#{s} #{arr[1]}"
+  puts n
+}
+```
+
+Here is the  `_` example written with *curly braces syntax* and in one line:
+
+```crystal-play
+def other_method
+  yield 42, "Hello", [1, "Crystal", 3]
+end
+
+other_method { |_, _, arr| puts arr[1] }
+```
+
+The main difference between using `do ... end` and `{ ... }` is how they bind the call:
+
+- `do ... end` binds to left-most call
+- `{ ... }` binds to the right-most call
+
+```crystal-play
+def generate_number
+  42
+end
+
+def with_double(n : Int32)
+  yield n * 2
+end
+
+with_double generate_number do |n|
+  puts n
+end
+
+# Same as:
+with_double(generate_number) do |n|
+  puts n
+end
+```
+
+But with `curly braces syntax`:
+
+```crystal
+def generate_number
+  42
+end
+
+def with_double(n : Int32)
+  yield n * 2
+end
+
+with_double generate_number { |n| puts n } # Error: 'generate_number' is not expected to be invoked with a block, but a block was given
+```
+
+The error is because with *curly braces* we are writing: `with_double(generate_number { |n| puts n })` instead of `with_double(generate_number) { |n| puts n }`
```

Comment:
```suggestion
The error is because with curly braces we are writing: `with_double(generate_number { |n| puts n })` instead of `with_double(generate_number) { |n| puts n }`
```